R/zchunk_L102.GDP.R

```diff
@@ -42,121 +121,204 @@ module_socioeconomics_L102.GDP_DISABLED <- function(command, ...) {
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L101.Pop_thous_R_Yh <- get_data(all_data, "L101.Pop_thous_R_Yh")
+    L101.Pop_thous_Scen_R_Yfut <- get_data(all_data, "L101.Pop_thous_Scen_R_Yfut")
+
+    ## iso--region lookup without extraneous data.  We'll use this several times.
+    iso_region32_lookup <- select(iso_GCAM_regID, iso, GCAM_region_ID)
+
+    ## Add region IDs to historical GDP and aggregate by region. Hang onto the country-level data
+    ## because we will use them again when we make the IMF adjustments
+    gdp_mil90usd_ctry <-
+        left_join_error_no_match(L100.gdp_mil90usd_ctry_Yh, iso_region32_lookup,
+                                 by='iso') %>%
+          rename(gdp = value)
+    gdp_mil90usd_rgn <- gdp_mil90usd_ctry %>%
+        group_by(GCAM_region_ID, year) %>% summarise(gdp=sum(gdp))
+    ## gdp_mil90usd_ctry:  iso, GCAM_region_ID, year, gdp
+    ## gdp_mil90usd_rgn:  GCAM_region_ID, year, gdp
+
+    ## Get the future GDP in the SSP scenarios.  These, it turns out, are PPP
+    ## values in 2005 dollars
+    gdp_bilusd_rgn_Yfut <-
+      filter(SSP_database_v9,MODEL=='OECD Env-Growth' & VARIABLE=='GDP|PPP') %>%
+      standardize_iso('REGION') %>% change_iso_code('rou','rom') %>%
+      left_join_error_no_match(iso_region32_lookup, by='iso') %>%
+      protect_integer_cols %>%
+      select_if(function(x){!any(is.na(x))}) %>% # apparently the SSP database has some missing in it; filter these out.
+      unprotect_integer_cols %>%
+      select(-MODEL, -iso, -VARIABLE, -UNIT) %>%
+      gather(year, gdp, -SCENARIO, -GCAM_region_ID) %>%
+      mutate(year = as.integer(year), gdp = as.numeric(gdp),
+             scenario=substr(SCENARIO, 1, 4)) %>% # Trim the junk off the end of
+                                        # the scenario names, leaving us with
+                                        # just SSP1, SSP2, etc.
+      group_by(scenario, GCAM_region_ID, year) %>%
+      summarise(gdp = sum(gdp)) %>%
+      select(scenario, GCAM_region_ID, year, gdp)
+    ## Units are billions of 2005$
+
+
+    gdp_bilusd_rgn_Yfut[['scenario']] <- substr(gdp_bilusd_rgn_Yfut[['scenario']], 1, 4)
+    gdp.mil90usd.SSP.rgn.yr <- join.gdp.ts(gdp_mil90usd_rgn, gdp_bilusd_rgn_Yfut, 'GCAM_region_ID')
+
+    ## Get the IMF GDP growth rates.  Some countries are missing, so we have to
+    ## add them in with an assumed zero growth rate.
+    imfgdp.growth <-
+        select(IMF_GDP_growth, one_of(c('ISO', IMF_FUTURE_YEARS))) %>%
+          standardize_iso('ISO') %>% change_iso_code('rou', 'rom') %>%
+          tidyr::gather(year, gdp.rate, -iso) %>%
```

Comment:
Are you sure?  When I first wrote this function it choked on `gather` calls without the namespace qualifier.  Have we added tidyr as an imported namespace since then?